Objects/typeobject.c

```diff
@@ -163,7 +163,17 @@ static void
 static_builtin_state_init(PyInterpreterState *interp, PyTypeObject *self)
 {
     if (!static_builtin_index_is_set(self)) {
-        static_builtin_index_set(self, interp->types.num_builtins_initialized);
+        assert(_Py_IsMainInterpreter(interp));
```

Comment:
In `static_builtin_state_clear()` there's a check for this rather than an assert. Maybe it should be an `if` here too, for symmetry? (Maybe the assert should be `!static_builtin_index_is_set()`?)

Reply:
Good point.  I'll flip it around.